### PR TITLE
cleanup: Modify pom files for java-pubsublite to enable errorprone analysis and general clean up

### PR DESCRIPTION
**Pull request by dpcollins-google**

### CODE REVIEW

File: google-cloud-pubsublite/pom.xml [hunk 1]
```diff
@@ -195,10 +183,6 @@
           </execution>
         </executions>
       </plugin>
-      <plugin>
-        <groupId>org.codehaus.mojo</groupId>
-        <artifactId>flatten-maven-plugin</artifactId>
-      </plugin>
```
Comment (anguillanneuf) [inline google-cloud-pubsublite/pom.xml:201]: Pinging our flatten maven plugin expert @stephaniewang526 here. Does flatten maven plugin conflict with errorprone?

Comment (stephaniewang526) [reply google-cloud-pubsublite/pom.xml:201]: No, it shouldn't. I would not recommend removing it.

Comment (dpcollins-google) [reply google-cloud-pubsublite/pom.xml:201]: This isn't removed, just moved to the plugins section on the parent POM

Comment (stephaniewang526) [reply google-cloud-pubsublite/pom.xml:201]: Okay as long as you're okay with generating flattened poms for parent/root poms as well; this is fine.

**Review by stephaniewang526 — APPROVED** [review]

**Review by anguillanneuf — APPROVED** [review]

**Pull request merged**